code/modules/food/plates/_plate.dm

```diff
@@ -0,0 +1,69 @@
+/obj/item/plate
+	name                = "plate"
+	desc                = "A small plate, suitable for serving food."
+	material            = /decl/material/solid/stone/ceramic
+	icon                = 'icons/obj/food/plates/small_plate.dmi'
+	icon_state          = ICON_STATE_WORLD
+	material_alteration = MAT_FLAG_ALTERATION_COLOR | MAT_FLAG_ALTERATION_NAME
+	w_class             = ITEM_SIZE_SMALL
+	var/is_dirty
+
+/obj/item/plate/Destroy()
+	if(istype(loc, /obj/item/chems/food))
+		var/obj/item/chems/food/food = loc
+		if(food.plate == src)
+			food.plate = null
+	return ..()
+
+/obj/item/plate/proc/make_dirty(obj/item/chems/food/food)
+	if(!is_dirty)
+		is_dirty = food?.filling_color || COLOR_WHITE
+		update_icon()
+
+/obj/item/plate/clean(clean_forensics = TRUE)
+	. = ..()
+	if(is_dirty)
+		is_dirty = null
+		update_icon()
+
+/obj/item/plate/on_update_icon()
+	. = ..()
+	if(is_dirty)
+		var/image/I = image(icon, "[icon_state]-dirty")
+		I.appearance_flags |= RESET_COLOR
+		I.color = is_dirty
+		add_overlay(I)
+
+/obj/item/plate/attackby(obj/item/W, mob/living/user)
+	// Plating food.
+	if(istype(W, /obj/item/chems/food))
+		if(!user.try_unequip(W))
```

Comment:
Maybe move this bloc into a proc which can be also called by food attackby? Seems like it's pretty much same code as there.